subsys/mgmt/mcumgr/transport/src/smp_lorawan.c

```diff
@@ -0,0 +1,263 @@
+/*
+ * Copyright (c) 2024, Jamie McCrae
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/lorawan/lorawan.h>
+#include <zephyr/mgmt/mcumgr/smp/smp.h>
+#include <zephyr/mgmt/mcumgr/transport/smp.h>
+#include <zephyr/mgmt/mcumgr/mgmt/handlers.h>
+
+#include <mgmt/mcumgr/transport/smp_internal.h>
+#include <mgmt/mcumgr/transport/smp_reassembly.h>
+
+LOG_MODULE_REGISTER(smp_lorawan, CONFIG_MCUMGR_TRANSPORT_LORAWAN_LOG_LEVEL);
+
+static void smp_lorawan_downlink(uint8_t port, bool data_pending, int16_t rssi, int8_t snr,
+				 uint8_t len, const uint8_t *hex_data);
+
+static int smp_lorawan_uplink(struct net_buf *nb);
+
+static uint16_t smp_lorawan_get_mtu(const struct net_buf *nb);
+
+static struct lorawan_downlink_cb lorawan_smp_downlink_cb = {
+	.port = CONFIG_MCUMGR_TRANSPORT_LORAWAN_FRAME_PORT,
+	.cb = smp_lorawan_downlink
+};
+
+struct smp_transport smp_lorawan_transport = {
+	.functions.output = smp_lorawan_uplink,
+	.functions.get_mtu = smp_lorawan_get_mtu,
+};
+
+#ifdef CONFIG_SMP_CLIENT
+struct smp_client_transport_entry smp_lorawan_client_transport = {
+	.smpt = &smp_lorawan_transport;
+	.smpt_type = SMP_LORAWAN_TRANSPORT;
```

Comment:
for my own education, why `;` here? Or maybe it's a typo?

Reply:
Typo, should be `,`, fixed